.gitmodules

```diff
@@ -1,3 +1,4 @@
 [submodule "storage"]
 	path = storage
 	url = https://github.com/Azure/azure-storage-go
+	branch = v0.1.0
```

Comment:
The name `branch` is misleading here. See this thread:
http://stackoverflow.com/questions/19986075/git-submodule-to-track-remote-branch

Alternatively, try cloning my fork to ensure it works :D